python-package/xgboost/testing/__init__.py

```diff
@@ -429,8 +429,8 @@ def make_categorical(
     categories = np.arange(0, n_categories)
     for col in df.columns:
         if rng.binomial(1, cat_ratio, size=1)[0] == 1:
-            df.loc[:, col] = df[col].astype("category")
-            df.loc[:, col] = df[col].cat.set_categories(categories)
+            df[col] = df[col].astype("category")
+            df[col] = df[col].cat.set_categories(categories)
```

Comment:
Per https://pandas.pydata.org/docs/user_guide/basics.html#astype, we should not mix `astype()` with `loc()` accessor, as `loc()` tries to fit in what we are assigning to the current dtypes. So `df.loc[:, col] = df[col].astype("category")` would have no effect.